vnext/ReactUWP/Modules/NativeUIManager.cpp

```diff
@@ -112,6 +112,33 @@ winrt::XamlRoot NativeUIManager::tryGetXamlRoot()
   return nullptr;
 }
 
+XamlView NativeUIManager::reactPeerOrContainerFrom(winrt::FrameworkElement fe)
```

Comment:
>reactPeerOrContainerFrom [](start = 26, length = 24)

Logic in function is similar but not the same as TouchEventHandler::TagFromOriginalSource, can share code between these?  (or getReactId which is the only thing calling this)
Given the other place doesn't use m_host / not sure why it was needed?

Reply:
Native control may have and set tag too and it's doesn't have react tag. For example, customer make NavigationView as a new ViewManager, but NavigationViewItem is not. and they may use NavigationView like this. NavigationViewItem can take focus and be source of keyboard, but we can't cast it to matched View.

<NavigationView>
<NavigationViewItem Icon="Accept" Tag="1" />
<NavigationViewItem Icon="Reject" Tag="2" />
</NavigationView

I expect TagFromOriginalSource would be removed and replace with this function, but it should not in this PR.


---
In reply to: [295387588](https://github.com/microsoft/react-native-windows/pull/2623#discussion_r295387588) [](ancestors = 295387588)